src/components/BackupRestoreModal.tsx

```diff
@@ -21,16 +24,14 @@ const BackupRestoreModal = () => {
   const setShowToast = useSetRecoilState(displayToast);
   const setLastAction = useSetRecoilState(lastAction);
 
-  const backupData = async () => {
-    const file = await db.export({ prettyJson: true });
-    const blob = new Blob([file], { type: "application/json" });
-    const url = URL.createObjectURL(blob);
-    const a = document.createElement("a");
-    a.href = url;
-    a.download = `ZinZenBackup-${new Date().toLocaleDateString()}.json`;
-    document.body.appendChild(a);
-    a.click();
-    document.body.removeChild(a);
+  const exportData = async (strategy: ExportStrategy) => {
```

Comment:
since you have implemented the classes for all the strategy, i would recommend you convert this fn into a manager class that will allow you to use any strategy.
 `
export class ExportManager {
  private strategy: ExportStrategy;

  constructor(strategy: ExportStrategy) {
    this.strategy = strategy;
  }

  setStrategy(strategy: ExportStrategy): void {
    this.strategy = strategy;
  }

  async export(options?: Record<string, any>): Promise<void> {
    await this.strategy.export(options);
  }
}
`